cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/ExecutableExecutionData.java

```diff
@@ -217,6 +224,35 @@ public void finishExecutable(@Param(ScoreLangConstants.RUN_ENV) RunEnvironment r
         }
     }
 
+    public void canExecute(@Param(ScoreLangConstants.RUN_ENV) RunEnvironment runEnv,
+                           @Param(EXECUTION_RUNTIME_SERVICES) ExecutionRuntimeServices executionRuntimeServices,
+                           @Param(ScoreLangConstants.NODE_NAME_KEY) String nodeName,
+                           @Param(ScoreLangConstants.NEXT_STEP_ID_KEY) Long nextStepId) {
+        try {
+            if (!StringUtils.isEmpty(runEnv.getExecutionPath().getParentPath())) {
+                //If it is start of a sub flow then the check should not happen
+                runEnv.putNextStepPosition(nextStepId);
+                return;
+            }
+
+            while (!executionPreconditionService.canExecute(valueOf(executionRuntimeServices.getExecutionId()))) {
+                try {
+                    logger.warn("Execution precondition not fulfilled. Waiting for it to be true.");
+                    Thread.sleep(5_000);
+                } catch (InterruptedException e) {
+                    logger.error("Thread was interrupted while waiting for execution precondition to be fulfilled.");
+                }
+            }
+
+            // put the next step position for the navigation
+            runEnv.putNextStepPosition(nextStepId);
+        } catch (RuntimeException e) {
+            logger.error("There was an error running the canExecute execution step of: \'" + nodeName +
```

Comment:
remove \n\t
also make sure you don't log this error 2 times (you log it, you throw it, I assume somewhere you catch and log it again)